docs/module-api.md

```diff
@@ -0,0 +1,40 @@
+# Module API Review Notes
+
+## Current Major Limitations
+- Only local state is supported
+- Only the `aws-base` module works so far. Module type aliases do not work, so it must still be referenced as `aws-base`.
+- Only the `apply` command works. To clean up, run `terraform destroy`
+- Multiple opta files (i.e. environments or peer opta files) are not supported
+
```

Comment:
What do you think should be the milestones to reach deprecation of the module API?
If you can add it in this document it will give some context about the next steps.
Also see my comment about the TODO to link them to these milestones.